Library/Homebrew/extend/os/mac/system_config.rb

```diff
@@ -28,14 +28,8 @@ def describe_xquartz
       "#{MacOS::XQuartz.version} => #{describe_path(MacOS::XQuartz.prefix)}"
     end
 
-    def describe_system_ruby
-      s = ""
-      case RUBY_VERSION
-      when /^1\.[89]/, /^2\.0/
-        s << "#{RUBY_VERSION}-p#{RUBY_PATCHLEVEL}"
-      else
-        s << RUBY_VERSION
-      end
+    def describe_homebrew_ruby
+      s = describe_homebrew_ruby_version
```

Comment:
@xu-cheng `describe_homebrew_ruby_version` will possibly return a frozen string. The following lines may conditionally modify `s`. Adding a `.dup` would be great. :-)